mutable_tree.go

```diff
@@ -26,14 +28,16 @@ var ErrVersionDoesNotExist = errors.New("version does not exist")
 //
 // The inner ImmutableTree should not be used directly by callers.
 type MutableTree struct {
-	*ImmutableTree                  // The current, working tree.
-	lastSaved      *ImmutableTree   // The most recently saved tree.
-	orphans        map[string]int64 // Nodes removed by changes to working tree.
-	versions       map[int64]bool   // The previous, saved versions of the tree.
-	allRootLoaded  bool             // Whether all roots are loaded or not(by LazyLoadVersion)
-	ndb            *nodeDB
+	*ImmutableTree                                  // The current, working tree.
+	lastSaved                *ImmutableTree         // The most recently saved tree.
+	orphans                  map[string]int64       // Nodes removed by changes to working tree.
+	versions                 map[int64]bool         // The previous, saved versions of the tree.
+	allRootLoaded            bool                   // Whether all roots are loaded or not(by LazyLoadVersion)
+	unsavedFastNodeAdditions map[string]*FastNode   // FastNodes that have not yet been saved to disk
```

Comment:
Why the unsavedFastNodeAdditions and unsavedFastNodeRemovals do not need mtx to protect?  the map would not be concurrent access?